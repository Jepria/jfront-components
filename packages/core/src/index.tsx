export * from '@jfront-ui/app'
export * from '@jfront-ui/datepicker'
export * from '@jfront-ui/form'
//export * from '@jfront-ui/grid'
export * from '@jfront-ui/hooks'
//export * from '@jfront-ui/input'
export * from '@jfront-ui/layout'
//export * from '@jfront-ui/pagingbar'
//export * from '@jfront-ui/tabpanel'
//export * from '@jfront-ui/toolbar'
