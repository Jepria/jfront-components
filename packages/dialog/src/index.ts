export * from './SimpleDialog'
export * from './AlertDialog'
export * from './FormDialog'
export * from './ErrorDialog'
export * from './WarningDialog'