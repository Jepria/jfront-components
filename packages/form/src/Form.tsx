import React, { ForwardRefExoticComponent, RefAttributes } from "react"
import { Label, LabelProps } from "@jfront/ui-label"
import styled from "styled-components"

const StyledForm = styled.form`
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  display: flex;
  flex-direction: column;
  padding-left: 10px;
  padding-top: 5px;
  font-family: Arial, sans-serif;
  font-size: small;
  overflow: auto;
`

const StyledField = styled.div`
  -webkit-box-sizing: border-box;
  box-sizing: border-box;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  margin-bottom: 1em;
  @media (max-width: 575px) {
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
  }
`

const StyledFieldControl = styled.div`
  flex-direction: column;
  box-sizing: border-box;
  display: inline-flex;
  -webkit-box-flex: 1;
  -ms-flex: 1 1 0;
  flex: 1 1 0;
`

export interface FormFieldSetProps
  extends React.FieldsetHTMLAttributes<HTMLFieldSetElement> {
  legend?: string
  renderLegend?: (legend?: string) => React.ReactNode
}

export interface FormLabelProps extends LabelProps {
  required?: boolean
}

const StyledFormLabel = styled(Label)<FormLabelProps>`
  min-width: 150px;
  max-width: 200px;
  width: 100%;
  line-height: 1.5715;
  @media (max-width: 575px) {
    justify-content: flex-start;
    text-align: left;
    min-width: unset;
    max-width: unset;
    width: unset;
  }
  ${(props) =>
    props.required
      ? `&::before{
            display: inline-block;
            margin-right: 4px;
            color: #ff4d4f;
            font-size: 14px;
            font-family: SimSun, sans-serif;
            line-height: 1;
            content: '*';
            vertical-align: top;
          }`
      : ""}
`

const StyledFieldSet = styled.fieldset`
  border: 1px solid rgba(81, 162, 238, 1);
  border-radius: 5px;
`

const StyledLegend = styled.legend`
  color: rgba(81, 162, 238, 1);
  margin: 0 5px;
`

export interface FormControlProps extends React.HTMLAttributes<HTMLDivElement> {
  error?: string
}

export interface FormFieldProps extends React.HTMLAttributes<HTMLDivElement> {
  label?: string
  required?: boolean
  renderLabel?: (label?: string, required?: boolean) => React.ReactNode
}

type Form = ForwardRefExoticComponent<
  React.FormHTMLAttributes<HTMLFormElement> & RefAttributes<HTMLFormElement>
> & {
  Field: React.FC<FormFieldProps>
  Label: React.FC<FormLabelProps>
  Legend: React.FC<React.HTMLAttributes<HTMLLegendElement>>
  Control: React.FC<FormControlProps>
  FieldSet: React.FC<FormFieldSetProps>
}

const InternalForm = React.forwardRef<
  HTMLFormElement,
  React.FormHTMLAttributes<HTMLFormElement>
>((props, ref) => {
  return (
    <StyledForm {...props} ref={ref}>
      {props.children}
    </StyledForm>
  )
})

const Form: Form = InternalForm as Form

Form.Field = (props: FormFieldProps) => {
  return (
    <StyledField {...props}>
      {props.label && !props.renderLabel && (
        <Form.Label required={props.required}>{props.label}</Form.Label>
      )}
      {props.renderLabel && props.renderLabel(props.label, props.required)}
      {props.children}
    </StyledField>
  )
}

Form.Label = (props: FormLabelProps) => {
  return <StyledFormLabel {...props}>{props.children}</StyledFormLabel>
}

Form.Legend = (props: React.HTMLAttributes<HTMLLegendElement>) => {
  return <StyledLegend {...props}>{props.children}</StyledLegend>
}

Form.FieldSet = (props: FormFieldSetProps) => {
  return (
    <StyledFieldSet {...props}>
      {props.legend && !props.renderLegend && (
        <Form.Legend>{props.legend}</Form.Legend>
      )}
      {props.renderLegend && props.renderLegend(props.legend)}
      {props.children}
    </StyledFieldSet>
  )
}

Form.Control = (props: FormControlProps) => {
  return (
    <StyledFieldControl {...props}>
      {React.Children.only(props.children)}
      {props.error && (
        <span
          style={{
            fontFamily: "tahoma, arial, helvetica, sans-serif",
            fontSize: 11,
            color: "red",
          }}
        >
          {props.error}
        </span>
      )}
    </StyledFieldControl>
  )
}

export { Form }
