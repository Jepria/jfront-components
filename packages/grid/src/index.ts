export * from './common'
export * from './jepgrid'
