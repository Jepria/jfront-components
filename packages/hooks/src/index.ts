export * from "./useOnClickOutside"
export * from "./useRouteMapping"
export * from "./useTree/"
export * from "./useTree/types"
export * from "./useFilter"
