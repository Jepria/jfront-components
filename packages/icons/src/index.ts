export * from "./Add"
export * from "./Delete"
export * from "./Edit"
export * from "./Exclamation"
export * from "./First"
export * from "./Last"
export * from "./Loading"
export * from "./Next"
export * from "./OpenIcon"
export * from "./Prev"
export * from "./Refresh"
export * from "./Save"
export * from "./Search"
export * from "./SettingsSmall"
export * from "./Split"
export * from "./View"
