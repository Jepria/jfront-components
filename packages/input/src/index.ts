export interface InputProps {
  error?: string
  name?: string
  isLoading?: boolean
}

export * from "./SelectInput"
export * from "./TextInput"
export * from "./NumberInput"
export * from "./MaskedInput"
export * from "./MoneyInput"
export * from "./DecimalInput"
export * from "./MoneyInput"
