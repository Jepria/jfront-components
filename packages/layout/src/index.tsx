import styled from 'styled-components';

export const FormContainer = styled.div`
  padding: 5px;
`;

export const VerticalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const HorizontalLayout = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
      -ms-flex-direction: row;
          flex-direction: row;
`;

export const Page = styled.div`
  box-sizing: border-box;
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
`;

export const Header = styled.header`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;

export const Content = styled.section`
  box-sizing: border-box;
  width: 100%;
  overflow: auto;
  -webkit-box-flex: 1;
      -ms-flex: 1;
          flex: 1;
  -ms-flex-positive: 1;
      flex-grow: 1;
`;

export const Footer = styled.footer`
  box-sizing: border-box;
  height: auto;
  width: 100%;
`;
