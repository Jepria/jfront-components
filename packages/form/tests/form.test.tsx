import { render } from "@testing-library/react"
import React from "react"
import { Form } from "../src"

test("Checking for the existence of an element Form, FormField", () => {
  render(<Form />)
})

test("Matches snapshot ", () => {
  const { asFragment } = render(
    <>
      <Form>
        <Form.Field>
          <Form.Label required>Text input:</Form.Label>
          <Form.Control>
            <input type="text" style={{ display: "inline-flex" }} />
          </Form.Control>
        </Form.Field>
        <Form.Field>
          <Form.Label>Select input</Form.Label>
          <Form.Control>
            <select style={{ display: "inline-flex" }}>
              <option value={1}>one</option>
              <option value={2}>two</option>
              <option value={3}>three</option>
            </select>
          </Form.Control>
        </Form.Field>
      </Form>
    </>,
  )
  expect(asFragment()).toMatchSnapshot()
})

test("Form required renders correctly", () => {
  const asFragment = render(
    <>
      <Form>
        <Form.Field>
          <Form.Label required>Select input</Form.Label>
          <Form.Control>
            <select style={{ display: "inline-flex" }}>
              <option value={1}>one</option>
              <option value={2}>two</option>
              <option value={3}>three</option>
            </select>
          </Form.Control>
        </Form.Field>
      </Form>
    </>,
  )
  expect(asFragment.asFragment()).toMatchSnapshot()
})

test("Form isError renders correctly", () => {
  const asFragment = render(
    <>
      <Form>
        <Form.Field>
          <Form.Label>Select input</Form.Label>
          <Form.Control error="Error message">
            <select style={{ display: "inline-flex" }}>
              <option value={1}>one</option>
              <option value={2}>two</option>
              <option value={3}>three</option>
            </select>
          </Form.Control>
        </Form.Field>
      </Form>
    </>,
  )
  expect(asFragment.asFragment()).toMatchSnapshot()
})

test("Form layout renders correctly", () => {
  const tools = render(
    <Form.FieldSet name="fields" legend="row">
      <Form.Field style={{ flexBasis: "50%" }}>
        <Form.Label>Text input:</Form.Label>
        <Form.Control>
          <input type="text" style={{ display: "inline-flex" }} />
        </Form.Control>
      </Form.Field>
      <Form.Field style={{ flexBasis: "50%" }}>
        <Form.Label>Select input</Form.Label>
        <Form.Control>
          <select style={{ display: "inline-flex" }}>
            <option value={1}>one</option>
            <option value={2}>two</option>
            <option value={3}>three</option>
          </select>
        </Form.Control>
      </Form.Field>
    </Form.FieldSet>,
  )
  expect(tools.asFragment()).toMatchSnapshot()
})
