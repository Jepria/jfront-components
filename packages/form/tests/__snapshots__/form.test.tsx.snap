// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form isError renders correctly 1`] = `
<DocumentFragment>
  <form
    class="sc-AxirZ fWRIjU"
  >
    <div
      class="sc-AxiKw kAZRMG"
    >
      <label
        class="sc-AxjAm DLLwe sc-AxhUy bUIkeS"
      >
        Select input
      </label>
      <div
        class="sc-AxhCb bgYvQa"
      >
        <select
          style="display: inline-flex;"
        >
          <option
            value="1"
          >
            one
          </option>
          <option
            value="2"
          >
            two
          </option>
          <option
            value="3"
          >
            three
          </option>
        </select>
        <span
          style="font-family: tahoma, arial, helvetica, sans-serif; font-size: 11px; color: red;"
        >
          Error message
        </span>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Form layout renders correctly 1`] = `
<DocumentFragment>
  <fieldset
    class="sc-AxgMl eyCrfk"
    name="fields"
  >
    <legend
      class="sc-AxheI kOhSQw"
    >
      row
    </legend>
    <div
      class="sc-AxiKw kAZRMG"
      style="flex-basis: 50%;"
    >
      <label
        class="sc-AxjAm DLLwe sc-AxhUy bUIkeS"
      >
        Text input:
      </label>
      <div
        class="sc-AxhCb bgYvQa"
      >
        <input
          style="display: inline-flex;"
          type="text"
        />
      </div>
    </div>
    <div
      class="sc-AxiKw kAZRMG"
      style="flex-basis: 50%;"
    >
      <label
        class="sc-AxjAm DLLwe sc-AxhUy bUIkeS"
      >
        Select input
      </label>
      <div
        class="sc-AxhCb bgYvQa"
      >
        <select
          style="display: inline-flex;"
        >
          <option
            value="1"
          >
            one
          </option>
          <option
            value="2"
          >
            two
          </option>
          <option
            value="3"
          >
            three
          </option>
        </select>
      </div>
    </div>
  </fieldset>
</DocumentFragment>
`;

exports[`Form required renders correctly 1`] = `
<DocumentFragment>
  <form
    class="sc-AxirZ fWRIjU"
  >
    <div
      class="sc-AxiKw kAZRMG"
    >
      <label
        class="sc-AxjAm DLLwe sc-AxhUy gnUDJm"
        required=""
      >
        Select input
      </label>
      <div
        class="sc-AxhCb bgYvQa"
      >
        <select
          style="display: inline-flex;"
        >
          <option
            value="1"
          >
            one
          </option>
          <option
            value="2"
          >
            two
          </option>
          <option
            value="3"
          >
            three
          </option>
        </select>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`Matches snapshot  1`] = `
<DocumentFragment>
  <form
    class="sc-AxirZ fWRIjU"
  >
    <div
      class="sc-AxiKw kAZRMG"
    >
      <label
        class="sc-AxjAm DLLwe sc-AxhUy gnUDJm"
        required=""
      >
        Text input:
      </label>
      <div
        class="sc-AxhCb bgYvQa"
      >
        <input
          style="display: inline-flex;"
          type="text"
        />
      </div>
    </div>
    <div
      class="sc-AxiKw kAZRMG"
    >
      <label
        class="sc-AxjAm DLLwe sc-AxhUy bUIkeS"
      >
        Select input
      </label>
      <div
        class="sc-AxhCb bgYvQa"
      >
        <select
          style="display: inline-flex;"
        >
          <option
            value="1"
          >
            one
          </option>
          <option
            value="2"
          >
            two
          </option>
          <option
            value="3"
          >
            three
          </option>
        </select>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
